assets/scripts/lobby/chat4.js

```diff
@@ -286,67 +288,115 @@ function addToRoomChat(data) {
+                var highlightChatColour = "";
+                var setHighlightColorToYellow = $(".setHighlightColorsToYellow")[0].checked;
 
-                    //if its a server text or special text
-                    if (data[i].classStr && data[i].classStr !== "") {
-                        str = "<li class='" + data[i].classStr + " " + addClass + "'><span class='date-text'>" + date + "</span> " + filteredMessage;
+                // console.log("true?"  + selectedChat[data[i].username]);
+
+                if (selectedChat[data[i].username] === true) {
+                    if (setHighlightColorToYellow === true) {
+                        highlightChatColour = "#ffff9e";
                     }
-                    //its a user's chat so put some other stuff on it
                     else {
-                        str = "<li class='" + addClass + "'><span style='background-color: " + highlightChatColour + "' username='" + data[i].username + "'><span class='date-text'> " + date + "</span> <span class='username-text'>" + data[i].username + ":</span> " + filteredMessage + "</span></li>";
+                        highlightChatColour = docCookies.getItem("player" + getIndexFromUsername(data[i].username) + 'HighlightColour');
                     }
+                }
 
-                    //if they've muted this player, then just dont show anything. reset str to nothing.
-                    if(isPlayerMuted(data[i].username) === true){
-                        str = "";
-                    }
+                //if its a server text or special text
+                if (data[i].classStr && data[i].classStr !== "") {
+                    str = "<li class='" + data[i].classStr + " " + addClass + "'><span class='date-text'>" + date + "</span> " + unquotedMessage;
+                }
+                //its a user's chat so put some other stuff on it
+                else {
+                    str = "<li class='" + addClass + "'><span style='background-color: " + highlightChatColour + "' username='" + data[i].username + "'><span class='date-text'> " + date + "</span> <span class='username-text'>" + data[i].username + ":</span> " + (unquotedMessage || "<i>Quoting:</i>") + "</span></li>";
+                }
 
-                    $(".room-chat-list").append(str);
-                    scrollDown("room-chat-room");
-                    scrollDown("room-chat-room2");
+                //if they've muted this player, then just dont show anything. reset str to nothing.
+                if (isPlayerMuted(data[i].username) === true) {
+                    str = "";
                 }
 
-                //else if there is a '[' character, then assume the user is quoting a chunk of text
-                else{
-                    var strings = filteredMessage.split("[");
+                $(".room-chat-list").append(str);
+                scrollDown("room-chat-room");
+                scrollDown("room-chat-room2");
```

Comment:
scrollDowns are being called in line 400 and 401? Perhaps move these out of the if blocks and put them once at the end?